sormas-backend/src/main/java/de/symeda/sormas/backend/deletionconfiguration/CoreEntityDeletionService.java

```diff
@@ -0,0 +1,75 @@
+package de.symeda.sormas.backend.deletionconfiguration;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.ejb.EJB;
+import javax.ejb.LocalBean;
+import javax.ejb.Singleton;
+import javax.inject.Inject;
+
+import de.symeda.sormas.api.utils.DateHelper;
+import de.symeda.sormas.backend.caze.CaseFacadeEjb;
+import de.symeda.sormas.backend.contact.ContactFacadeEjb;
+import de.symeda.sormas.backend.event.EventFacadeEjb;
+import de.symeda.sormas.backend.event.EventParticipantFacadeEjb;
+import de.symeda.sormas.backend.immunization.ImmunizationFacadeEjb;
+import de.symeda.sormas.backend.travelentry.TravelEntryFacadeEjb;
+
+@LocalBean
+@Singleton
+public class CoreEntityDeletionService {
+
+	private final List<EntityTypeFacadePair> coreEntityFacades = new ArrayList<>();
+
+	@EJB
+	private DeletionConfigurationService deletionConfigurationService;
+
+	public CoreEntityDeletionService() {
+	}
+
+	@Inject
+	public CoreEntityDeletionService(
+		CaseFacadeEjb.CaseFacadeEjbLocal caseFacadeEjb,
+		ContactFacadeEjb.ContactFacadeEjbLocal contactFacadeEjb,
+		EventFacadeEjb.EventFacadeEjbLocal eventFacadeEjb,
+		EventParticipantFacadeEjb.EventParticipantFacadeEjbLocal eventParticipantFacadeEjb,
+		ImmunizationFacadeEjb.ImmunizationFacadeEjbLocal immunizationFacadeEjb,
+		TravelEntryFacadeEjb.TravelEntryFacadeEjbLocal travelEntryFacadeEjb) {
+		coreEntityFacades.add(EntityTypeFacadePair.of(CoreEntityType.CASE, caseFacadeEjb));
+		coreEntityFacades.add(EntityTypeFacadePair.of(CoreEntityType.CONTACT, contactFacadeEjb));
+		coreEntityFacades.add(EntityTypeFacadePair.of(CoreEntityType.EVENT, eventFacadeEjb));
+		coreEntityFacades.add(EntityTypeFacadePair.of(CoreEntityType.EVENT_PARTICIPANT, eventParticipantFacadeEjb));
+		coreEntityFacades.add(EntityTypeFacadePair.of(CoreEntityType.IMMUNIZATION, immunizationFacadeEjb));
+		coreEntityFacades.add(EntityTypeFacadePair.of(CoreEntityType.TRAVEL_ENTRY, travelEntryFacadeEjb));
+	}
+
+	public void executeAutomaticDeletion() {
+
+		coreEntityFacades.forEach(coreEntityType -> {
+			DeletionConfiguration coreEntityTypeConfig = deletionConfigurationService.getCoreEntityTypeConfig(coreEntityType.coreEntityType);
+
+			if (coreEntityTypeConfig.deletionReference!=null && coreEntityTypeConfig.deletionPeriod != null) {
+				Date referenceDeletionDate = DateHelper.subtractDays(new Date(), coreEntityTypeConfig.deletionPeriod);
```

Comment:
This logic seems a bit unintuitive for me, and I actually think that calling the result of this calculation referenceDeletionDate is incorrect. Wouldn't it make more sense to pass the deletionPeriod to executeAutomaticDeletion, and then do the calculation in that method?